relay-common/src/constants.rs

```diff
@@ -478,29 +478,32 @@ impl fmt::Display for FractionUnit {
     }
 }
 
+const CUSTOM_UNIT_SIZE: usize = 15;
+
 /// Custom user-defined units without builtin conversion.
 #[derive(Clone, Copy, Eq, PartialEq, Hash)]
-pub struct CustomUnit([u8; 15]);
+pub struct CustomUnit([u8; CUSTOM_UNIT_SIZE]);
 
 impl CustomUnit {
     /// Parses a `CustomUnit` from a string.
     pub fn parse(s: &str) -> Result<Self, ParseMetricUnitError> {
-        if s.len() > 15 || !s.is_ascii() {
+        if s.len() > CUSTOM_UNIT_SIZE || !s.is_ascii() {
             return Err(ParseMetricUnitError(()));
         }
 
         let mut unit = Self(Default::default());
-        unit.0.copy_from_slice(s.as_bytes());
+        unit.0[0..s.len()].copy_from_slice(s.as_bytes());
```

Comment:
```suggestion
        unit.0[..s.len()].copy_from_slice(s.as_bytes());
```

you can also assume the array is already zero-filled

Reply:
> you can also assume the array is already zero-filled

Ah, because of the default initialization, right? I made the construction explicit now to make that more clear.